provider/blob/src/static_data_provider.rs

```diff
@@ -2,48 +2,51 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use crate::blob_schema::BlobSchema;
+use crate::path_util;
 use icu_provider::{
     prelude::*,
     serde::{SerdeDeDataProvider, SerdeDeDataReceiver},
 };
-use litemap::LiteMap;
-use serde::Deserialize;
-
-const STATIC_STR_DATA: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/static_data.bincode"));
+use serde::de::Deserialize;
 
 /// A data provider loading data statically baked in to the binary. Useful for testing in situations
 /// where setting up a filesystem is tricky (e.g. WASM).
 ///
 /// This should probably not be used in production code since it bloats the binary.
 pub struct StaticDataProvider {
-    json: LiteMap<&'static str, &'static str>,
+    blob: BlobSchema<'static>,
```

Comment:
potential followup: StaticDataProvider should probably eventually be named BlobDataProvider with a lifetime, where SDP is just when you construct it with a `'static` lifetime

Reply:
I will respond to this (and add better docs tests) soon.

Reply:
Follow-up: https://github.com/unicode-org/icu4x/issues/848

I want to keep it as StaticDataProvider for the time being since I don't completely know yet what the signature of BlobDataProvider will be.